modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/store/Store.es.js

```diff
@@ -0,0 +1,256 @@
+				if (isDispatching) {
+					throw new Error(
+						'Illegal unsubscribe() call during dispatch'
+					);
+				}
+
+				subscribers.delete(nextSubscriberID);
+			};
+		},
+	};
+
+	return store;
+}
+
+/**
+ * Takes an object where the keys and values are namespaces and their
+ * corresponding reducer functions, and returns a single reducer function.
+ *
+ * See: https://redux.js.org/api/combinereducers
+ */
+export function combineReducers(reducers) {
+	const namespacesAndReducers = Object.entries(reducers);
+
+	return function combinedReducer(state, action) {
+		let hasChanged = false;
+
+		const nextState = {};
+
+		for ([namespace, reducer] of namespacesAndReducers) {
+			const previous = state[namespace];
+			const next = reducer(previous, action);
+
+			if (typeof next === 'undefined') {
+				throw new Error(`Reducer for ${namespace} returned undefined`);
+			}
+
+			hasChanged = hasChanged || previous !== next;
+
+			nextState[namespace] = next;
+		}
+
+		return hasChanged ? nextState : state;
+	};
+}
+
+export function defaultReducer(state, _action) {
+	return state;
+}
+
+/**
+ * Takes an array of reducers and returns a single reducer that runs them
+ * serially.
+ *
+ * This produces a "flat" state, unlike `combineReducers`, which produces a
+ * "nested/namespaced" state.
+ *
+ * See: https://github.com/redux-utilities/reduce-reducers
+ */
+export function reduceReducers(reducers) {
+	return function reducedReducer(state, action) {
+		return reducers.reduce((currentState, reducer, DELETE_ME) => {
+			const nextState = reducer(currentState, action);
+
+			if (typeof nextState === 'undefined') {
+				throw new Error(`Reducer for returned undefined`);
+			}
+
+			return nextState;
+		}, state);
+	};
+}
+
+const STORES = {};
+
+const SUB_QUEUE = {};
+
+export default {
+	Util: {
+		combineReducers,
+		reduceReducers,
+	},
+
+	get(name) {
+		return STORES[name];
+	},
+
+	subscribe(name, subscriber) {
+		if (STORES[name]) {
+			STORES[name].subscribe(subscriber);
+		} else {
+			if (SUB_QUEUE[name]) {
+				SUB_QUEUE[name].push(subscriber);
+			} else {
+				SUB_QUEUE[name] = [subscriber];
+			}
+		}
+	},
+
+	register(name, reducer, preloadedState, enhancer) {
+		if (!name || typeof name !== 'string') {
+			throw new Error('`name` must be a non-empty string');
+		}
+
+		if (name in STORES) {
+			throw new Error(`Store "${name}" is already registered`);
+		}
+
+		let currentReducer = reducer || defaultReducer;
+
+		const baseStore = createStore(currentReducer, preloadedState, enhancer);
+
+		const store = {
+			...baseStore,
+
+			replaceReducer(nextReducer) {
+				currentReducer = nextReducer;
+
+				baseStore.replaceReducer(nextReducer);
+			},
+		};
+
+		STORES[name] = {
+			get reducer() {
+				return currentReducer;
+			},
+
+			get store() {
+				return store;
+			},
+		};
+
+		if (SUB_QUEUE[name]) {
+			SUB_QUEUE[name].forEach((subscriber) =>
+				store.subscribe(subscriber)
+			);
+
+			delete SUB_QUEUE[name];
+		}
```

Comment:
This is simply a check to see if any other JS is waiting to be subscribed to the given store, if there is, we then attach those subscribers.